### PR TITLE
Extracts code from Ember and sets minimum ember-cli-babel

### PR DESCRIPTION
**Pull request by locks**
- [x] Add Code
- [x] Add tests
- [x] Fix `loc`
- [x] Deprecate `loc`
- [x] Bump `ember-cli-babel` to version that supports `@ember/string`

### CODE REVIEW

**Comment by rwjblue** [comment]
ATM I think the tests are probably using `Ember.String` and not our addon/index.js versions (due to babel polyfill for modules).

**Comment by dwickern** [comment]
Where'd `htmlSafe` / `isHTMLSafe` go?

**Comment by locks** [comment]
[`@ember/template`](https://github.com/emberjs/rfcs/blob/master/text/0236-deprecation-ember-string.md#emberstringhtmlsafe-and-emberstringishtmlsafe)

**Comment by rwjblue** [comment]
@locks - status?

**Comment by locks** [comment]
@rwjblue cleaning all the relevant repos in the AM

File: addon/index.js [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+  @param {String} str The string to format
+  @param {Array} formats Optional array of parameters to interpolate into string.
+  @return {String} formatted string
+  @public
+*/
+export function loc(str, formats) {
+  deprecate(
+    'loc is deprecated, use an internationalization or localization addon instead.',
+    false,
+    { id: 'ember-string-utils.loc', until: '3.5.0', url: 'http://babeljs.io/docs/learn-es2015/#template-strings' }
```
Comment (SergeAstapov) [inline addon/index.js]: Should this be a link to ember-intl or ember-i18n or some other page about the topic?

**Comment by locks** [comment]
I'm updating `ember-cli-babel` and the `index.js` name since those are quick, and landing this.

**Pull request merged**